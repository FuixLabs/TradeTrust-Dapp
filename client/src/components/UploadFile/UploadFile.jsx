import React from "react";

export default function UploadFile({ handleOnFileChange }) {
  return (
    <div
      style={{
        marginTop: 20,
      }}
    >
      <h3>Upload your file</h3>
      <div>
        <input
          data-testid="fileReader"
          type="file"
          onChange={handleOnFileChange}
        />
      </div>
    </div>
  );
}
