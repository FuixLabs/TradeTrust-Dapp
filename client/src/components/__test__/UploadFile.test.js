import {render, screen, cleanup, fireEvent} from '@testing-library/react';
import rendered from 'react-test-renderer';
import UploadFile from '../UploadFile/UploadFile';

const fileReaderId = 'fileReader';

const setup = () => {
  render(<UploadFile />);
  const fileReader = screen.getByTestId(fileReaderId);
  return fileReader;
}

afterEach(() => {
  cleanup();
});

test('Should render UploadFile component', () => {
  render(<UploadFile />);
  const fileReaderElement = screen.getByTestId('fileReader');
  expect(fileReaderElement).toBeInTheDocument();
});

test('Input type of <input> should be file', () => {
  const fileReader = setup();
  expect(fileReader).toBeInTheDocument();
  expect(fileReader).toHaveAttribute("type", "file");
});

test('Matches snapshot', () => {
  const tree = rendered.create(<UploadFile />).toJSON();
  expect(tree).toMatchSnapshot();
});
