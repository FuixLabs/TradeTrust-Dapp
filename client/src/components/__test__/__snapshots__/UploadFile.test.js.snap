// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  style={
    Object {
      "marginTop": 20,
    }
  }
>
  <h3>
    Upload your file
  </h3>
  <div>
    <input
      data-testid="fileReader"
      type="file"
    />
  </div>
</div>
`;
